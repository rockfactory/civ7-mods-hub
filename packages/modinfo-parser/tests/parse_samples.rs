use modinfo_parser::Mod;

#[test]
fn parse_samples() {
    insta::glob!("samples/*.modinfo", |path| {
        let text = std::fs::read_to_string(path).unwrap();
        let parsed = Mod::parse(text.as_bytes()).unwrap();
        insta::with_settings!({ sort_maps => true }, {
            insta::assert_debug_snapshot!(parsed);
            let serialized = parsed.to_string().unwrap();
            let reparsed = Mod::parse(serialized.as_bytes()).unwrap();
            assert_eq!(parsed, reparsed);
        });
    });
}
