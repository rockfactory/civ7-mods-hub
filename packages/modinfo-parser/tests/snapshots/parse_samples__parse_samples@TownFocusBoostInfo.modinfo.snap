---
source: packages/modinfo-parser/tests/parse_samples.rs
expression: parsed
input_file: packages/modinfo-parser/tests/samples/TownFocusBoostInfo.modinfo
---
Mod {
    id: "YMD_TownFocusBoostInfo",
    version: Some(
        "1.1",
    ),
    properties: Properties {
        name: Some(
            "LOC_MOD_YMD_TOWNFOCUSBOOSTINFO",
        ),
        description: Some(
            "LOC_MOD_YMD_TOWNFOCUSBOOSTINFO_DESCRIPTION",
        ),
        authors: Some(
            "Yamada",
        ),
        package: Some(
            "Mod",
        ),
        affects_saved_games: false,
        special_thanks: None,
        version: None,
        compatibility: None,
        url: None,
    },
    dependencies: [
        Dependency {
            id: "base-standard",
            title: Some(
                "LOC_MODULE_BASE_STANDARD_NAME",
            ),
        },
    ],
    localized_text: [
        "text/en_us/ModuleText.xml",
        "l10n/ja_JP_Text.xml",
    ],
    action_criteria: [
        ActionCriterion {
            id: "always",
            always_met: true,
            age_in_use: None,
            mod_in_use: None,
        },
    ],
    action_groups: [
        ActionGroup {
            id: "town_focus-boost-info-actions",
            scope: Some(
                "game",
            ),
            criteria: Some(
                "always",
            ),
            properties: ActionGroupProperties {
                load_order: None,
            },
            actions: [
                ActionGroupActions {
                    import_files: [],
                    ui_scripts: [
                        ActionItem {
                            path: "script/TownFocusCityYields.js",
                            locale: None,
                        },
                        ActionItem {
                            path: "script/TownFocusWarehouse.js",
                            locale: None,
                        },
                        ActionItem {
                            path: "script/TownFocusBoostInfo.js",
                            locale: None,
                        },
                    ],
                    update_database: [],
                    update_text: [],
                },
            ],
        },
    ],
}
